### PR TITLE
fix: build error

### PR DESCRIPTION
**Pull request by ShawnLee2018**

### CODE REVIEW

**Comment by dna2github** [comment]
sorry, could you please share more details on what you are doing with this change?
and also could you please share the build error reproduce steps? thx!

**Comment by ShawnLee2018** [comment]
Issues1: `web/contrib_contents.go` method "contribRenderSearchResult" used the structure 'ResultInput' specifies the 'SearchOptions' values,  However, the ResultInput structure in `web/api.go` does not have this field when it is defined . 
for this issues I added the SearchOptions declaration in place of the ResultInput definition  

Issues2: The number of `func serveSearchErr` return values is only one in `web/server.go`, but  there got tow in the function implementation.  
for this issues I  changed the function 'serveSearchErr' return values number to 2.


Both of these issues cause the zoekt-webserver module to fail to compile.  My compilation environment is Ubuntu 20.04.4 LTS go version go1.17.7 linux/amd64

**Comment by dna2github** [comment]
thx for the details. and yes, sorry, i worked on my mac and it got weird that i can compile the program without error ...
the regression error is caused by rebase change (ownership: google/zoekt -> sourcegraph/zoekt, memory usage 5x -> 1x)

could you please keep `ResultInput` unchange (keep sourcegraph/zoekt version) and just remove the line of `SearchOptions: "",` in `web/contrib_contents.go` (in sourcegraph/zoekt, the SearchOptions now is used as a standalone struct `zoekt.SearchOptions`)? i think we need to fix `contribRenderSearchResult` in another commit to make sure it is synced up with current renderer.

it will make sure the build pased; and i will merge your commit then. thx!

File: web/api.go [hunk 1]
```diff
@@ -35,12 +35,13 @@ type LastInput struct {
...
-	Last        LastInput
-	QueryStr    string
-	Query       string
-	Stats       zoekt.Stats
-	Duration    time.Duration
-	FileMatches []*FileMatch
+	Last          LastInput
+	QueryStr      string
+	Query         string
+	SearchOptions string
```
Comment (dna2github) [inline web/api.go]: remove it (keep sourcegraph/zoekt version) and remove the line where it is `SearchOptions: ""` in `web/contrib_contents.go`. thx!

Comment (ShawnLee2018) [reply web/api.go]: thx for your review. I have revised it according to your suggestions.

**Comment by dna2github** [comment]
thx. and sorry again; just found that the fix is already in my local with `git add`ed ...

**Comment by ShawnLee2018** [comment]
Thank you for your review. I have revised it according to your suggestions.  

> thx. and sorry again; just found that the fix is already in my local with `git add`ed ...

that's okay. As long as it's fixed.

**Comment by dna2github** [comment]
merged yours.

**Pull request merged**